www/csp-violation-report-endpoint.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+// Start configure
+$log_file = dirname(__FILE__) . '/csp-violations.log';
+$log_file_size_limit = 1000000; // bytes - once exceeded no further entries are added
+$email_address = 'slockwood@catchpoint.com';
+$email_subject = 'Content-Security-Policy violation';
+// End configuration
+
+$current_domain = preg_replace('/www\./i', '', $_SERVER['SERVER_NAME']);
+$email_subject = $email_subject . ' on ' . $current_domain;
+
+http_response_code(204); // HTTP 204 No Content
+
+$json_data = file_get_contents('php://input');
+
+// We pretty print the JSON before adding it to the log file
+if ($json_data = json_decode($json_data)) {
+    $json_data = json_encode($json_data, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES);
+
+    if (!file_exists($log_file)) {
+        // Send an email
+        $message = "The following Content-Security-Policy violation occurred on " .
+            $current_domain . ":\n\n" .
+            $json_data .
+            "\n\nFurther CPS violations will be logged to the following log file, but no further email notifications will be sent until this log file is deleted:\n\n" .
+            $log_file;
+        mail(
```

Comment:
Does this use the main (blocking) process? I'm worried about having 2000 users on the site and all of these going off at the same time.